pkg/connect/client/datastore/v1alpha1/fake/fake.go

```diff
@@ -0,0 +1,145 @@
+}
+
+// ListNodeSelectors implements v1alpha1.DataStoreClient.
+func (c *fakeDataStoreClient) ListNodeSelectors(ctx context.Context, req *datastorev1alpha1.ListNodeSelectorsRequest) (*datastorev1alpha1.ListNodeSelectorsResponse, error) {
+	c.fake.Mu.Lock()
+	defer c.fake.Mu.Unlock()
+	nodes := c.fake.AttestedNodes
+	var selectors []*datastorev1alpha1.ListNodeSelectorsResponse_NodeSelectors
+	for _, node := range nodes {
+		newNode := proto.Clone(node).(*datastorev1alpha1.AttestedNode)
+		selectors = append(selectors, &datastorev1alpha1.ListNodeSelectorsResponse_NodeSelectors{
+			SpiffeId:  newNode.GetSpiffeId(),
+			Selectors: newNode.GetSelectors(),
+		})
+	}
+
+	return &datastorev1alpha1.ListNodeSelectorsResponse{NodeSelectors: selectors}, nil
+}
+
+func (c *fakeDataStoreClient) CountAttestedNodes(ctx context.Context, req *datastorev1alpha1.CountAttestedNodesRequest) (*datastorev1alpha1.CountAttestedNodesResponse, error) {
+	c.fake.Mu.Lock()
+	defer c.fake.Mu.Unlock()
+
+	count := int32(len(c.fake.AttestedNodes))
+	return &datastorev1alpha1.CountAttestedNodesResponse{Count: count}, nil
+}
+
+func (c *fakeDataStoreClient) CreateAttestedNode(ctx context.Context, req *datastorev1alpha1.CreateAttestedNodeRequest) (*datastorev1alpha1.CreateAttestedNodeResponse, error) {
+	c.fake.Mu.Lock()
+	defer c.fake.Mu.Unlock()
+	node := proto.Clone(req.Node).(*datastorev1alpha1.AttestedNode)
+	node.Selectors = nil
+	c.fake.AttestedNodes[req.GetNode().GetSpiffeId()] = node
+	return &datastorev1alpha1.CreateAttestedNodeResponse{Node: node}, nil
+}
+
+func (c *fakeDataStoreClient) DeleteAttestedNode(ctx context.Context, req *datastorev1alpha1.DeleteAttestedNodeRequest) (*datastorev1alpha1.DeleteAttestedNodeResponse, error) {
+	c.fake.Mu.Lock()
+	defer c.fake.Mu.Unlock()
+
+	if _, exists := c.fake.AttestedNodes[req.GetSpiffeId()]; !exists {
+		return nil, status.Error(codes.NotFound, "node not found")
+	}
+
+	delete(c.fake.AttestedNodes, req.GetSpiffeId())
+	return &datastorev1alpha1.DeleteAttestedNodeResponse{}, nil
+}
+
+func (c *fakeDataStoreClient) FetchAttestedNode(ctx context.Context, req *datastorev1alpha1.FetchAttestedNodeRequest) (*datastorev1alpha1.FetchAttestedNodeResponse, error) {
+	c.fake.Mu.Lock()
+	defer c.fake.Mu.Unlock()
+
+	node, exists := c.fake.AttestedNodes[req.GetSpiffeId()]
+	if !exists {
+		return nil, status.Error(codes.NotFound, "node not found")
+	}
+	return &datastorev1alpha1.FetchAttestedNodeResponse{Node: proto.Clone(node).(*datastorev1alpha1.AttestedNode)}, nil
+}
+
+func (c *fakeDataStoreClient) UpdateAttestedNode(ctx context.Context, req *datastorev1alpha1.UpdateAttestedNodeRequest) (*datastorev1alpha1.UpdateAttestedNodeResponse, error) {
+	c.fake.Mu.Lock()
+	defer c.fake.Mu.Unlock()
+
+	_, exists := c.fake.AttestedNodes[req.Node.GetSpiffeId()]
+	if !exists {
+		return nil, status.Error(codes.NotFound, "node not found")
+	}
+	newNode := proto.Clone(req.Node).(*datastorev1alpha1.AttestedNode)
+	newNode.Selectors = nil
```

Comment:
The SPIRE SQL store only updates fields in the AttestedNodeMask. I think we can ignore the mask itself here, but we shouldn't clear the selectors.

Reply:
eyes like an 🦅